ERCS/erc-7986.md

```diff
@@ -0,0 +1,882 @@
+
+    function registerSessionKey(
+        bytes32 sessionKey,
+        uint256 expiration
+    ) external;
+
+    function updateMerkleRoot(
+        bytes32 newRoot,
+        uint256 newLeafCount
+    ) external;
+}
+```
+
+#### IRevocationManager
+
+Handles capability token revocation with super-function authority.
+
+```solidity
+interface IRevocationManager {
+    struct RevocationRecord {
+        bytes32 capabilityPointer;
+        uint256 revokedBlock;
+        address revokedBy;
+        bytes32 reason;
+        bool active;
+    }
+
+    event CapabilityRevoked(
+        bytes32 indexed capabilityPointer,
+        address indexed revokedBy,
+        bytes32 reason
+    );
+
+    event RevocationAuthorityUpdated(
+        address indexed newAuthority,
+        address indexed oldAuthority
+    );
+
+    function revokeCapability(
+        bytes32 capabilityPointer,
+        bytes32 reason
+    ) external;
+
+    function isRevoked(bytes32 capabilityPointer)
+        external view returns (bool);
+
+    function getRevocationRecord(bytes32 capabilityPointer)
+        external view returns (RevocationRecord memory);
+
+    function updateRevocationAuthority(address newAuthority) external;
+}
+```
+
+### Transient Storage Integration
+
+The framework utilizes **[EIP-1153](eip-1153) transient storage** for gas-efficient temporary data management:
+
+```solidity
+library TransientCapabilityStorage {
+    // Transient storage slots for capability validation
+    bytes32 constant ACTIVE_TOKEN_SLOT = keccak256("capability.active.token");
+    bytes32 constant VALIDATION_TAG_SLOT = keccak256("capability.validation.tag");
+    bytes32 constant NONCE_TRACKING_SLOT = keccak256("capability.nonce.tracking");
+
+    function setActiveToken(bytes32 tokenHash) internal {
+        assembly {
+            tstore(ACTIVE_TOKEN_SLOT, tokenHash)
+        }
+    }
+
+    function getActiveToken() internal view returns (bytes32 tokenHash) {
+        assembly {
+            tokenHash := tload(ACTIVE_TOKEN_SLOT)
+        }
+    }
+
+    function setValidationTag(bytes32 tag) internal {
+        assembly {
+            tstore(VALIDATION_TAG_SLOT, tag)
+        }
+    }
+
+    function getValidationTag() internal view returns (bytes32 tag) {
+        assembly {
+            tag := tload(VALIDATION_TAG_SLOT)
+        }
+    }
+
+    function markNonceUsed(uint256 nonce) internal {
+        bytes32 slot = keccak256(abi.encode(NONCE_TRACKING_SLOT, nonce));
+        assembly {
+            tstore(slot, 1)
+        }
+    }
+
+    function isNonceUsed(uint256 nonce) internal view returns (bool used) {
+        bytes32 slot = keccak256(abi.encode(NONCE_TRACKING_SLOT, nonce));
+        assembly {
+            used := tload(slot)
+        }
+    }
+}
+```
+
+### Framework Modifier
+
+Protected functions use the capability enforcement modifier:
+
+```solidity
+modifier onlyWithCapability(bytes32 capabilityPointer, bytes calldata tokenData) {
```

Comment:
Similarly, modifiers are also normally implementation details. The specification section should only describe the externally visible behaviours of a contract, not how it is implemented.